Windows and broad search only alternative approach

KEMAL-YILDIRIM opened the pull request:
Hi, 

Apologies if this is inconvenient, this is my first contribution on GitHub.
I have been using your pluging for a while.
Since I am working on scattered c# projects with many libraries,
I did some changes to be able to select the solution in a different way.
I have also added a checkhealth.

I don't know if this is a useful thing for the sake of your intentions for this plugin but, 
I thought it would be good to share my findings. 
Which let me work in windows environment a bit nicer.
I haven't test it in linux btw.

thanks for all you have done.

seblyng commented:
Could you explain a bit what you have changed and why? What is this trying to fix? What is not working for you now that this fixes?

KEMAL-YILDIRIM commented:
Problem is similar like this #146 
Almost all the time I am working with 20+ projects and libraries, where most of the projects have their own sln files on similar parent directory levels.
It's a bit tidious changing the current directory restarting everything and trying to pick the correct sln
this allows me to start from the current buffer location, see all the sln and projects and pick whichever I want when broad search enabled.
I have figured when I change the schedule with coroutine it's faster for broad search option so I have changed that as well
I have added some predefined folder names to ignore searching like node_modules, .git ...
It shouldn't be introducing any change other than the broad search but as I said I haven't tested on linux.

seblyng commented:
I am just a bit confused, because I just looked a bit at the code and it looks very similar as to how it works today. Also, reading your description on how you want it to work seems like (how I at least thought) it is working already.

Do you have a more concrete example of where it's not able to find solutions when broad_search is enabled? I added tests for this a while ago, so I would really like to know a test scenario that would fail with the current solution

KEMAL-YILDIRIM commented:
Sorry about that, you are absolutely right, my mistake. Apperantly I didn't have the new version somehow. It looks great. Sorry again. Closing this one.

seblyng commented:
Okay perfect🙌 nice to know that it works good for you!
